### PR TITLE
Various fixes

### PR DESCRIPTION
**Pull request by jspaaks**
Various of small fixes, like

- removed ``label`` argument from similarity function
- added argument ``known_adducts`` to ``derive_ionmode``
- added caching of ``load_adducts`` function result (#144)
- made user workflow integration test more challenging to pass

### CODE REVIEW

**Comment by jspaaks** [comment]
@florian-huber could you check if the filters I'm using in ``integration-tests/test_user_workflow.py`` make sense (also wrt the order in which they are called, and the parameterization)? Other comments also welcome of course.

File: matchms/filtering/derive_ionmode.py [hunk 1]
```diff
@@ -19,8 +19,9 @@ def derive_ionmode(spectrum_in):
 
     spectrum = spectrum_in.clone()
 
-    # Load lists of known adducts
-    known_adducts = load_adducts()
+    if known_adducts is None:
```
Comment (florian-huber) [inline matchms/filtering/derive_ionmode.py]: I was confused by the default of ``known_adducts=None``, which made me expect that the function knows no adducts (I already wanted to complain that this seems no good default ...). Would it be possible to use another flag, maybe ``known_adducts='default'``?

**Review by florian-huber — CHANGES_REQUESTED** [review]
The integration test looks good @jspaaks, also the used filters (and their order) !
I just left one small remark about a the default for ``add_adduct()`` (nothing close to my heart, so feel free to override). 
Nice fix with the caching of the adducts loading by the way.

File: matchms/filtering/derive_ionmode.py (hunk 1 above)
Comment (jspaaks) [reply matchms/filtering/derive_ionmode.py]: I changed it such that you can put in the filename if you want to use your own adducts file. Most users will want the default; they dont have to specify anything. Regardless of which they choose, the caching will still work so performance is unaffected.

Comment (florian-huber) [reply matchms/filtering/derive_ionmode.py]: That solves it for me, thanks!

**Review by florian-huber — APPROVED** [review]

**Pull request merged**